Mathlib/MeasureTheory/Function/LpSpace/DomAct/Basic.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2023 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.MeasureTheory.Integral.Bochner
+import Mathlib.MeasureTheory.Function.AEEqFun.DomAct
+
+/-!
+# Action of `Mᵈᵐᵃ` on `Lᵖ` spaces
+
+In this file we define action of `Mᵈᵐᵃ` on `MeasureTheory.Lp E p μ`
+given by `mk c • ⟨⟨f, _⟩, _⟩ = ⟨⟨(f <| c • ·), _⟩, _⟩`.
+We also prove basic properties of this action.
+-/
+
+open MeasureTheory Filter
+open scoped ENNReal
+
+namespace DomMulAct
+
+variable {M N α E : Type _} [MeasurableSpace M] [MeasurableSpace N]
+  [MeasurableSpace α] [NormedAddCommGroup E] {μ : MeasureTheory.Measure α} {p : ℝ≥0∞}
+
+section SMul
+
+variable [SMul M α] [SMulInvariantMeasure M α μ] [MeasurableSMul M α]
+
+@[to_additive]
+instance : SMul Mᵈᵐᵃ (Lp E p μ) where
+  smul c f := Lp.compMeasurePreserving (mk.symm c • ·) (measurePreserving_smul _ _) f
+
+@[to_additive (attr := simp)]
+theorem smul_Lp_val (c : Mᵈᵐᵃ) (f : Lp E p μ) : (c • f).1 = c • f.1 := rfl
+
+@[to_additive]
+theorem smul_Lp_ae_eq (c : Mᵈᵐᵃ) (f : Lp E p μ) : c • f =ᵐ[μ] (f <| mk.symm c • ·) :=
+  Lp.coeFn_compMeasurePreserving _ _
+
+@[to_additive]
+theorem mk_smul_toLp (c : M) {f : α → E} (hf : Memℒp f p μ) :
+    mk c • hf.toLp f =
+      (hf.comp_measurePreserving <| measurePreserving_smul c μ).toLp (f <| c • ·) :=
+  rfl
+
+@[to_additive (attr := simp)]
+theorem smul_Lp_const [IsFiniteMeasure μ] (c : Mᵈᵐᵃ) (a : E) :
+    c • Lp.const p μ a = Lp.const p μ a :=
+  rfl
+
+instance [SMul N α] [SMulCommClass M N α] [SMulInvariantMeasure N α μ] [MeasurableSMul N α] :
+    SMulCommClass Mᵈᵐᵃ Nᵈᵐᵃ (Lp E p μ) :=
+  Subtype.val_injective.smulCommClass (fun _ _ ↦ rfl) fun _ _ ↦ rfl
+
+instance [NormedRing 𝕜] [Module 𝕜 E] [BoundedSMul 𝕜 E] : SMulCommClass Mᵈᵐᵃ 𝕜 (Lp E p μ) :=
+  Subtype.val_injective.smulCommClass (fun _ _ ↦ rfl) fun _ _ ↦ rfl
+
+instance [NormedRing 𝕜] [Module 𝕜 E] [BoundedSMul 𝕜 E] : SMulCommClass 𝕜 Mᵈᵐᵃ (Lp E p μ) :=
+  .symm _ _ _
+
+-- We don't have a typeclass for additive versions of the next few lemmas
+-- Should we add `AddDistribAddAction` with `to_additive` both from `MulDistribMulAction`
+-- and `DistribMulAction`?
```

Comment:
It looks to me like this comment has shifter to the wrong place. Can you put it right above the `DistribSmul`?

Reply:
Oh, nevermind.